geo/src/algorithm/simplify_vw.rs

```diff
@@ -6,20 +6,21 @@ use crate::{
 use std::cmp::Ordering;
 use std::collections::BinaryHeap;
 
+use rstar::primitives::CachedEnvelope;
 use rstar::{RTree, RTreeNum};
 
-/// Store triangle information
-// current is the candidate point for removal
+/// Store triangle information. Area is used for ranking in the priority queue and determining removal
 #[derive(Debug)]
 struct VScore<T, I>
 where
     T: CoordFloat,
 {
     left: usize,
+    /// The current [Point] index in the original [LineString]: The candidate for removal
     current: usize,
     right: usize,
     area: T,
-    // `visvalingam_preserve` uses `intersector`, `visvalingam` does not
+    // `visvalingam_preserve` uses `intersector`, `visvalingam` does not, so it's always false
     intersector: I,
```

Comment:
IIRC `intersector` is now always a `bool` and the `I` generic param has become inexplicable. Can we remove it, or is there a benefit to keeping this struct generic?